### PR TITLE
Added migration to split enrollment name into first and last

### PR DESCRIPTION
**Pull request by aoby**
Part 1 of 2 from refactoring https://github.com/code-dot-org/code-dot-org/pull/11672

### CODE REVIEW

File: dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class AddFirstAndLastNamesToPdEnrollments < ActiveRecord::Migration[5.0]
+  class Pd::Enrollment < ActiveRecord::Base
+  end
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb:3]: Why is this here?

File: dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  def change
+    change_table(:pd_enrollments) do |t|
+      t.string :first_name
+      t.string :last_name
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb]: Does it make sense to add these after `name`?

File: dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb (hunk 1 above)
Comment (aoby) [reply dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb:3]: Good question. Removing...

File: dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    reversible do |dir|
+      dir.up do
+        Pd::Enrollment.find_each do |enrollment|
+          name_split = enrollment.name.split(' ', 2)
+          enrollment.first_name = name_split[0]
+          enrollment.last_name = name_split[1] || ''
+          enrollment.save!
+        end
+      end
+    end
```
Comment (davidsbailey) [inline dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb:21]: Do we have some guarantee that this migration will never get re-run? If not, it might be nice (and easy) to only populate first and last name from `name` if first and last name are not already set.

**Comment by aoby** [comment]
:( Now tests are failing since these new fields are required (not null) in SQL but not implemented in the model. @ashercodeorg what's your preferred solution?
1. 2 commits (migration and model) in the same PR so tests pass
2. modify the factory so tests pass, but still real-world enrollments will fail to save for the same reason
3. merge both PRs in rapid succession, despite failures in this one (my least favorite option)

File: dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb (hunk 3 above)
Comment (aoby) [reply dashboard/db/migrate/20161104075248_add_first_and_last_names_to_pd_enrollments.rb:21]: It would fail on lines 6-8 above if run again because the first_name and last_name columns already exist.

**Comment by ashercodeorg** [comment]
My preference, then, would be to (a) run the migration creating and populating `first_name` and `last_name`, making those fields optional and `full_name` optional, (b) add the model, and (c) run a migration making the `first_name` and `last_name` required and removing `full_name`.

Those three ((a), (b), and (c)) happening in different DTPs.

**Comment by aoby** [comment]
I think you mean `name` (the old DB field). `full_name` is a convenience method I added (that I want to keep) to handle splitting and recombining `first_name` and `last_name`. We want to keep `name` around for a little while as a backup, so I'll defer (c) and in the meantime allow `first_name` and `last_name` to accept NULL in SQL, but require their presence in the model so we should have no NULL values.
